packages/valory/skills/apy_estimation_abci/behaviours.py

```diff
@@ -275,52 +292,130 @@ def _fetch_batch(self) -> Generator[None, None, None]:
             subgraph=self.context.fantom_subgraph,
         )
 
-        if fetched_block is None:
-            return
+        return fetched_block
 
-        # Fetch ETH price for block.
+    def _fetch_eth_price(
+        self, block: Dict[str, int]
+    ) -> Generator[None, None, Tuple[HttpMessage, Optional[float]]]:
+        """Fetch ETH price for block."""
         res_raw = yield from self.get_http_response(
             content=eth_price_usd_q(
                 self.context.spooky_subgraph.bundle_id,
-                fetched_block["number"],
+                block["number"],
             ),
             **self._spooky_api_specs,
         )
         res = self.context.spooky_subgraph.process_response(res_raw)
 
         eth_price = yield from self._handle_response(
             res,
-            res_context=f"ETH price for block {fetched_block}",
+            res_context=f"ETH price for block {block}",
             keys=("bundles", 0, "ethPrice"),
             subgraph=self.context.spooky_subgraph,
+            sleep_on_fail=False,
+        )
+
+        return res_raw, eth_price
+
+    def _check_non_indexed_block(
+        self, res_raw: HttpMessage
+    ) -> Generator[None, None, Optional[Tuple[Dict[str, int], float]]]:
+        """Check if we received a non-indexed block error and try to get the ETH price for the latest indexed block."""
+        res = self.context.spooky_subgraph.process_non_indexed_error(res_raw)
+        latest_indexed_block_error = yield from self._handle_response(
+            res,
+            res_context="indexing error that will be attempted to be handled",
+            keys=(0, "message"),
+            subgraph=self.context.spooky_subgraph,
+        )
+        if latest_indexed_block_error is None:
+            return None
+
+        match = re.match(NON_INDEXED_BLOCK_RE, latest_indexed_block_error)
+        if match is None:
+            return None
+
+        latest_indexed_block_number = match.group(1)
+        # we set the last digit to 0, so that it is more possible for the agents to reach to a consensus
+        # without requiring an extra round.
+        latest_indexed_block_number = int(
+            math.floor(float(latest_indexed_block_number) * 0.1) * 10
```

Comment:
this I don't understand. Why zero? And what if the last digit is already zero? We cannot do a `-1` because? 

I like your proper mathematical programmer approach. Here's the python way 
`int(str(latest_indexed_block_number)[:-1] + '0')`

I would suggest: latest_indexed_block_number - latest_indexed_block_number % 10

Reply:
We cannot do a `-1` because the relative difference of the results will not change.